src/mergeplg/merge.py

```diff
@@ -114,6 +118,194 @@ def calculate_cml_geometry(ds_cmls, disc=8):
+            y1_line=ds_cml.site_0_lat.data,
+            x2_line=ds_cml.site_1_lon.data,
+            y2_line=ds_cml.site_1_lat.data,
+            cml_id=ds_cml.cml_id.data,
+            x_grid=ds_rad.longitudes.data,
+            y_grid=ds_rad.latitudes.data,
+            grid_point_location=grid_location_radar,  
+        )
+        
+        # Calculate radar ranfall along CMLs        
+        self.R_rad = plg.spatial.get_grid_time_series_at_intersections(
+            grid_data=ds_rad.rainfall_amount,
+            intersect_weights=intersect_weights,
+        )
+        
+        # Store x and y grids for faster access later
+        self.xgrid, self.ygrid = ds_rad.xs.data, ds_rad.ys.data
+        
+class MergeAdditiveIDW(Merge):
+    def __init__(self, ds_cml, ds_rad, grid_location_radar='center', min_obs=5, smooth_factor=3):
+        Merge.__init__(self, ds_cml, ds_rad, grid_location_radar, min_obs, smooth_factor)
+        
+        # Calculate the difference between radar and CML
+        self.R_diff = xr.where(
+            (self.R_rad > 0) & (self.ds_cml.R > 0), 
+            self.ds_cml.R - self.R_rad, 
+            np.nan,
+        )
+        
+        self.R_diff['mid_x'] = (ds_cml.site_0_x + ds_cml.site_1_x) / 2
+        self.R_diff['mid_y'] = (ds_cml.site_0_y + ds_cml.site_1_y) / 2
+
+    def __call__(self, time, smooth_factor=1):
+        # Create array for storing interpolated values
+        return merge_additive_idw(
+            self.R_diff.sel(time = time), 
+            self.ds_rad.rainfall_amount.sel(time = time), 
+            where_rad=True, 
+            min_obs=5
+        )
+
+class MergeAdditiveBlockKriging(Merge):
+    def __init__(self, ds_cml, ds_rad, grid_location_radar='center', min_obs=5, smooth_factor=3, T = 0, drop_zero = True, kriging_param_rolling = 3, disc = 9):
+        Merge.__init__(self, ds_cml, ds_rad, grid_location_radar, min_obs, smooth_factor)
+        
+        # Calculate the difference between radar and CML
+        self.R_diff = xr.where(
+            (self.R_rad > 0) & (self.ds_cml.R > 0), 
+            self.ds_cml.R - self.R_rad, 
+            np.nan,
+        )
+        
+        # Midpoint of CML, used for variogram fitting
+        x_ = (ds_cml.site_0_x + ds_cml.site_1_x) / 2
+        y_ = (ds_cml.site_0_y + ds_cml.site_1_y) / 2
+        
+        
+        # Calculate CML geometry
+        self.x0 = calculate_cml_geometry(ds_cml, disc=disc)
+        
+        # Calculate kriging parameters
+        kriging_param = {}        
+        for time_mid in range(ds_cml.time.size):
+            x, y, values = [], [], []
+        
+            # Set lower and upper timesteps
+            lower = time_mid - T
+            if lower < 0:
+                lower = 0
+            upper = time_mid + T + 1
+            if upper > ds_cml.time.size:
+                upper = ds_cml.time.size
+        
+            # Timesteps in window
+            for time in ds_cml.time[lower:upper]:
```

Comment:
Just to make sure I understand correctly: This loop and the code below is used to also take data from earlier and later time steps into account when estimating the variogram, correct?

Reply:
Thats the idea, it might be a bit experimental. Is just a feature I wanted to test and it could be removed as it could add confusion.